yt/loaders.py

```diff
@@ -1268,90 +1268,102 @@ def flatten(l):
 
 
 # --- Loader for yt sample datasets ---
-# This utility will check to see if sample data exists on disc.
-# If not, it will download it.
-
-
-def load_sample(fn=None, specific_file=None, pbar=True):
+def load_sample(fn, progressbar: bool = True, **kwargs):
     """
-    Load sample data with yt. Simple wrapper around yt.load to include fetching
+    Load sample data with yt. Simple wrapper around `yt.load` to include fetching
     data with pooch.
 
+    The data registry table can be retrieved and visualized using
+    `yt.sample_data.api.get_data_registry_table`.
+
+    This function requires pandas and pooch to be installed.
+
     Parameters
     ----------
-    fn : str or None
-        The name of the sample data to load. This is generally the name of the
-        folder of the dataset. For IsolatedGalaxy, the name would be
-        `IsolatedGalaxy`.  If `None` is supplied, the return value
-        will be a list of all known datasets (by name).
-
-    specific_file : str, optional
-        optional argument -- the name of the file to load that is located
-        within sample dataset of `name`. For the dataset `enzo_cosmology_plus`,
-        which has a number of timesteps available, one may wish to choose
-        DD0003. The file specifically would be
-        `enzo_cosmology_plus/DD0003/DD0003`, and the argument passed to this
-        variable would be `DD0003/DD0003`
+    fn : str
+        The `filename` of the dataset to load, as defined in the data registry
+        table.
 
     pbar: bool
         display a progress bar
 
+    Any additional keyword argument is passed down to `yt.load`.
+    Note that in case of collision with predefined keyword arguments as set in
+    the data registry, the ones passed to this function take priority.
     """
 
-    fido = PoochHandle()
+    from yt.sample_data.api import (
+        _download_sample_data_file,
+        _get_archive_filename,
+        _get_test_data_dir_path,
+        get_data_registry_table,
+        lookup_on_disk_data,
+    )
 
-    if fn is None:
-        keys = []
-        for key in fido._registry:
-            for ext in _extensions_to_strip:
-                if key.endswith(ext):
-                    key = key[: -len(ext)]
-            keys.append(key)
-        return keys
+    pooch_logger = pooch.utilsget_logger()
```

Comment:
```suggestion
    pooch_logger = pooch.pooch.utils.get_logger()
```

Reply:
This is half correct. I'm indeed missing a `.`

Reply:
should be fixed now, I'll let you resolve